install_config/persistent_storage/storage_classes.adoc

```diff
@@ -0,0 +1,212 @@
+
+[options="header"]
+|===
+
+|Storage Type |Provisioner Plug-in Name
+
+|OpenStack Cinder
+|`kubernetes.io/cinder`
+
+|AWS Elastic Block Store (EBS)
+|`kubernetes.io/aws-ebs`
+
+|GCE Persistent Disk (gcePD)
+|`kubernetes.io/gce-pd`
+
+|GlusterFS
+|`kubernetes.io/glusterfs`
+
+
+
+|===
+
+
+[IMPORTANT]
+====
+For any chosen provisioner plug-ins, any relevant cloud, host or third-party configurations must also
+be set up, per provider required documentation.
+====
+
+[[defining-storage-classes]]
+== Defining a StorageClass
+
+StorageClasses are currently a globally scoped object and need to be created by cluster-admins or
+storage-admins.
+There are currently four plug-ins that are supported upstream. The following
+section will describe the basic Spec definition for a _StorageClass_ and specific examples for each of the supported plug-in types.
+
+[[basic-spec-defintion]]
+=== Basic StorageClass Spec Definition
+
+.StorageClass Basic Spec Definition
+====
+[source,yaml]
+----
+kind: StorageClass <1>
+apiVersion: storage.k8s.io/v1beta1 <2>
+metadata:
+  name: foo <3>
+provisioner: kubernetes.io/plug-in-type <4>
+parameters: <5>
+  param1: value
+  ...
+  paramN: value
+
+----
+<1> (required) The API object type.
+<2> (required) The current apiVersion.
+<3> (required) The name of the StorageClass.
+<4> (required) The type of provisioner associated with this storage class.
+<5> (optional) The parameters required for the specific provisioner, this will change
+from plug-in to plug-in.
+====
+
+
+[[openstack-cinder-spec]]
+=== OpenStack Cinder Spec
+
+.cinder-storageclass.yaml
+====
+[source,yaml]
+----
+kind: StorageClass
+apiVersion: storage.k8s.io/v1beta1
+metadata:
+  name: gold
+provisioner: kubernetes.io/cinder
+parameters:
+  type: fast  <1>
+  availability: nova <2>
+
+----
+<1> VolumeType created in Cinder. Default is empty.
+<2> Availability Zone. Default is empty.
+====
+
+[[aws-elasticblockstore-ebs]]
+=== AWS ElasticBlockStore (EBS)
+
+.aws-ebs-storageclass.yaml
+====
+[source,yaml]
+----
+kind: StorageClass
+apiVersion: storage.k8s.io/v1beta1
+metadata:
+  name: slow
+provisioner: kubernetes.io/aws-ebs
+parameters:
+  type: io1 <1>
+  zone: us-east-1d <2>
+  iopsPerGB: "10" <3>
+  encrypted: true <4>
+  kmsKeyId: keyvalue <5>
+
+----
+
+<1> io1, gp2, sc1, st1. See AWS docs for details. Default: gp2.
+<2> AWS zone. If not specified, a random zone from those where Kubernetes cluster has a node is chosen.
+<3> only for io1 volumes. I/O operations per second per GiB. AWS volume plug-in multiplies this with size of requested volume to compute IOPS of the volume and caps it at 20 000 IOPS (maximum supported by AWS, see AWS docs).
+<4> denotes whether the EBS volume should be encrypted or not. Valid values are true or false.
+<5> (optional) The full Amazon Resource Name of the key to use when encrypting the volume. If none is supplied but encrypted is true, a key is generated by AWS. See AWS docs for valid ARN value.
+====
```

Comment:
Should say Amazon Resourse Name (ARN) so the user knows what that is when you refer to it later